### ISSUE TITLE
Delete a pool without deleting its contents

### ISSUE THREAD

**Issue by andreasgeier** [bpmn-io/bpmn-js#1643]
__Is your feature request related to a problem? Please describe.__

I would like to be able to delete a pool without deleting all it's contents to simplify my diagram. Since containment rules forbid to drag the contents out of the pool onto the canvas, I'm forced to do it in XML or copy and paste into a new model.

__Describe the solution you'd like__

Keep the content of the pool when deleting the pool. 

Check out Miro's two-step approach, where the content of a deleted frame remains on the canvas and is selected. So the user can decide to keep it or remove it as well with a second delete action.

https://user-images.githubusercontent.com/938244/166635374-6153189d-5a3b-490b-976f-691cea0f4b7e.mov

__Additional context__

Requested by @falko via [Slack](https://camunda.slack.com/archives/CKGH9LR40/p1651643588005489), supported by @andreasgeier

**Comment by andreasgeier** [comment]
In case of multiple pools on the canvas, it is not possible to place elements outside of pools. Which means a problem for the solution described above. We could remove this limitation and handle it via linting rule to give the user more flexibility for modeling.

**Comment by nikku** [comment]
@andreasgeier Exactly. The difference between Miro and us is that elements outside of a pool are not permitted in many situations. I'd rather invest into "extract contents in new diagram" features (a refactoring action).

**Comment by andreasgeier** [comment]
I'd rather advocate becoming less rigid in modeling, now that we have linting in place as a solution to make the user aware of issues.

**Comment by nikku** [comment]
> I'd rather advocate becoming less rigid in modeling, now that we have linting in place as a solution to make the user aware of issues.

We are a domain specific editor, not a drawing tool cf. ["Editors, not drawing tools"](https://github.com/bpmn-io/design-principles#editors-not-drawing-tools). As such becoming less rigid (to the point of not being BPMN compliant anymore) means giving up a lot of our existing strength. At the moment, whatever is modeled is valid BPMN, which is one of our strong points.
